content/en/security/application_security/setup/go/dockerfile.md

```
@@ -0,0 +1,262 @@
+---
+title: Building your Go application with Datadog's WAF
+code_lang: docker
+type: multi-code-lang
+code_lang_weight: 10
+further_reading:
+- link: "/security/application_security/how-it-works/"
+  tag: "Documentation"
+  text: "How App and API Protection Works"
+- link: "/security/default_rules/?category=cat-application-security"
+  tag: "Documentation"
+  text: "OOTB App and API Protection Rules"
+- link: "/security/application_security/troubleshooting"
+  tag: "Documentation"
+  text: "Troubleshooting App and API Protection"
+---
+
+# Introduction
+
+App and API Protection for Go installation requirements can be abstract and the Go toolchain
+cross-compilation capabilities can make it hard to understand what has to be done precisely.
+
+In these cases, a more precise way to materialize these examples like a Dockerfile can be interesting.
+The goal of this guide is to be a step-by-step guide to a working Dockerfile, tailor-fitted for your usecase.
+
+## Walkthrough
+
+A lot of Dockerfiles found in this guide come from [appsec-go-test-app][4] repository. To try it out, first clone the repository:
+
+```sh
+git clone https://github.com/DataDog/appsec-go-test-app.git
+cd appsec-go-test-app
+```
+
+A list of `Dockerfile` examples can be found in the [`examples/docker`][3] directory.
+Here is an example of it in its simplest form:
+
+```dockerfile
+FROM golang:1 AS build
+WORKDIR /app
+COPY . .
+
+RUN go install github.com/DataDog/orchestrion@latest && \
+    orchestrion pin
+
+RUN orchestrion go build -o main .
+
+FROM ubuntu:noble
+COPY --from=build /app/main /usr/local/bin
+
+ENV DD_APPSEC_ENABLED=true
+ENTRYPOINT [ "/usr/local/bin/main" ]
+```
+
+This constitutes the simplest version of a working Dockerfile for a Go application with Datadog's WAF enabled.
+It would be better to run `orchestrion pin` ahead of time and commit it to your VCS for future builds, but this is not a strict requirement.
```

Cool, but that makes the "not so great" workflow be the 1st (and only) thing you show; and you're not even showing folks how to adjust their setup if they decide to go down this (better) route...

---

At first this was more of a [tutorial](https://diataxis.fr/tutorials/) than a [how-to](https://diataxis.fr/how-to-guides/) but I guess this frontier is blurry now, I will do as you say


```suggestion
If this is your first use of [orchestrion][5]. This dockerfile requires to run `orchestrion pin` beforehand and commit the resulting changes for it to work. Please refer to our [Getting Started for Go][15]
```